### ISSUE TITLE
SBO term for confidence score

### ISSUE THREAD

**Issue by draeger** [SBRG/ModelPolisher#5]
Request a new SBO term to be used for confidence scores.

**Comment by aebrahim** [comment]
How exactly will this work if SBO terms are to be unique per reaction?

**Comment by draeger** [comment]
At the moment there is no field where we can store the confidence scores. We either need parameters or something new. Parameters aren't appropriate, because we cannot refer to a reaction from them. Local parameters aren't suitable either because we would then need to create a kinetic law whose math element must not be empty. I am currently thinking about what to do with confidence scores.

**Comment by aebrahim** [comment]
Ah I see. That's still a TBD.

I think one approach would be to create an evidence type. So you could link to a paper, and classify the type of evidence it is. I'm not yet sure if that will work with every case though.

**Comment by draeger** [comment]
There is an SBML package for distributions, need to check if this can be helpful: http://sourceforge.net/p/sbml/code/HEAD/tree/trunk/specifications/sbml-level-3/version-1/distrib/sbml-level-3-distrib-package-proposal.pdf?format=raw

**Comment by aebrahim** [comment]
What parts in particular would be relevant? This seems to be about sampling from distributions, and I can't see how that's related.

**Comment by draeger** [comment]
Yes. I wanted to check if it also includes confidence scores, but haven't seen it either. Conclusion, we probably need some additional field where we can put this.

**Comment by aebrahim** [comment]
I think this calls for a new "citations" or "evidence" package

**Comment by draeger** [comment]
Good idea! I'll collect all other missing fields and see what else is needed. I'll raise this point in the next SBML team meeting (tomorrow).

**Comment by draeger** [comment]
This was further discussed in thread opencobra/schema/issues/4#issuecomment-317018249, where @matthiaskoenig had the idea to use more specific terms from the [evidenceontology.org](http://evidenceontology.org/browse/). We should check if we can make use of this here.

**Comment by draeger** [comment]
Here is an overview of the scores as these are usually defined in COBRA, where 0 is best and 4 is lowest confidence.
* 0 = Biochemical: Enzyme has been tested biochemically.
* 1 = Genetic: Gene overexpression and purification, gene deletions.
* 2 = Sequence: There is significant sequence similarity to another gene with known function.
* 3 = Physiological: There is physiological data to support inclusion in the model.
* 4 = Modeling: Reaction is included to improve simulation results

For the export from COBRA/BiGG models to SBML we will only need to find the closest terms from ECO for these 4 levels. For the other direction we will need to also define a rule how to match terms between those.

**Comment by matthiaskoenig** [comment]
And forgot:
About the rules: You just use the ontology tree to match the terms. I.e. everything which is below the respective terms is matched to the terms. If evidence codes in SBML not a subelement of the suggested ECOS than no match can be done.

**Comment by draeger** [comment]
This looks like a very good start! Thanks @matthiaskoenig. We should also direct @tpfau to this suggestion.

**Comment by matthiaskoenig** [comment]
Just to add to this:
The big advantage of using annotation via ECO is that it allows to store the confidence! and especially what is the basis of the confidence, because one can add multiple evidence annotations ! This is crucial for metabolic network reconstructions and one of the big short comings of the current confidence scores.

One wants to store for a reaction all the evidence which is there, not only the minimal common denominator.
Example given: One has a reaction R1
- there is some evidence based on homology to mouse (-> add an annotation to homology evidence)
- there is some evidence based on protein data (-> add an annotation to experimental evidence based on protein)
- there is some bioinformatics inference for R1 (-> add the inference evidence)
- there is some indirect evidence based on mRNA (-> add the infered from experimental data evidence)
If there is some evidence only in a certain tissue, based on omics data
- add a complex annotation of this evidence to the reaction.
Suddenly you have the **collection** of evidence and confidence for the reaction and not only a "0". Confidence scores is nothing anybody should use in a reconstruction in 2018.

**Comment by ChristianLieven** [comment]
Supporting this is something @midnighter and @cdiener may also want to consider when improving the [cobrapy](https://github.com/opencobra/cobrapy) parsers. Once this finds its way into Cobrapy.Model objects I'm very happy to start writing tests for this in [memote](https://github.com/opencobra/memote).

**Comment by tpfau** [comment]
In general I think using ECO here is a very good idea. 
However, there are methods which rely on the 0-4 schematic used by bigg, and we should offer some way to translate at least the ECO top levels:
```
        ECO:0000006 experimental evidence
        ECO:0000041 similarity evidence
        ECO:0000088 biological system reconstruction evidence
        ECO:0000177 genomic context evidence
        ECO:0000204 author statement
        ECO:0000212 combinatorial evidence
        ECO:0000311 imported information
        ECO:0000352 evidence used in manual assertion
        ECO:0000361 inferential evidence
        ECO:0000501 evidence used in automatic assertion
        ECO:0006055 high throughput evidence
```
To the 0-4 levels.

**Comment by Linelili** [comment]
Please note that COBRA's definition of the confidence scores (0= best, 4 = lowest confidence score) is inverse to the definition of Ines Thiele's and Bernhard Ø. Palsson's "A protocol for generating a high-quality genome-scale metabolic reconstruction", where 4 is the best and 0 the lowest confidence score (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125167/table/T2/?report=objectonly).

Hence, using ECO numbers instead of scores from 0 to 4 might help to avoid confusion.
